### DISCUSSION TITLE
Argo Rollout - ConfigMap, Secret Versioning - Stable/Canary Resource Deletion

### DISCUSSION THREAD

**Discussion by dlennon-ie** [argoproj/argo-rollouts#2802] [Q&A]
I'm looking to version my k8s resources by name by adding the container version into the name.
However the canary has deleted the previous ConfigMap & Secret, only the canary versions exist. I'm using helm to deploy my apps.
During testing I deleted my stable pod and it went into a broken state, it can't find it's version of ConfigMap & Secret.
Is there any way around this? Am I approach the problem from the wrong perspective?

**Comment by kostis-codefresh** [comment]
Could you elaborate a bit on what is your end goal? what exactly do you want to accomplish?

**Reply by dlennon-ie** [reply]
I want to have at least 2 configmaps available in my namespace, one for canary and one for stable. If we have cluster issues or scale elastically, the "new" stable nodes should use the "stable" configmap.
Out of the box I had my configmap use the same name in both and that led to issues of new deployments using the canary configmap.
I then added the image version as an id to the configmap name.
I then ran into issues during the canary rollout with the "stable" configmap being deleted.
I'm beginning to suspect it might be helm causing the clean up of resources - adding `"helm.sh/resource-policy": keep` stopped the deletion.
But this leads to manual management of configmaps - all doable but perhaps there is a better approach.

**Reply by kostis-codefresh** [reply]
The simplest and more trouble free approach it is to have a single confimap that is compatible with both versions (old and new). The old version will simply ignore entries it doesn't understand. So if for example you new version has a new property page_size=4, you just add it to the configmap and the old version doesn't do anything with it.

If you need more flexibility you should have a single configmap that has versions for properties. For example page_size_v1.3 ad page_size_v1.4. Each version (1.3 and 1.4) will read their own entries. Of course this needs collaboration with your developers.

Both of the above approaches use only one configmap. Would any of those would work in your case?

People have been successful with multiple configmaps but by using several hacks such as kustomize configmap generators. If you search Github discussions you will find several examples.

Example: https://github.com/argoproj/argo-rollouts/discussions/1930
